source/common/utils/logging.js

```diff
@@ -1,6 +1,78 @@
 // @flow
+import { isEmpty } from 'lodash';
+import type {
+  FormatMessageContextParams,
+  ConstructMessageBodyParams,
+  MessageBody
+} from '../types/logging.types';
+
+const DEFAULT_MESSAGE_BODY: MessageBody = {
+  at: '',
+  env: '',
+  ns: [
+    'daedalus',
+  ],
+  data: {},
+  app: [
+    'daedalus'
+  ],
+  msg: '',
+  pid: '',
+  sev: '',
+  thread: ''
+};
+
 export const stringifyData = (data: any) => JSON.stringify(data, null, 2);
 
 export const stringifyError = (error: any) => (
   JSON.stringify(error, Object.getOwnPropertyNames(error), 2)
 );
+
+export const formatContext = (context: FormatMessageContextParams): string => {
+  const { appName, electronProcess, level, network } = context;
+  return `[${appName}.*${network}*:${level}:${electronProcess}]`;
+};
+
+export const formatMessageTime = (date: Date): string => {
+  const [year, time] = date.toISOString().split('T');
+  return `[${year} ${time.slice(0, -1)} UTC]`;
+};
+
+export const constructMessageBody = (bodyData: ConstructMessageBodyParams): MessageBody => {
+  const { data = {} } = bodyData;
+  let messageBody = { ...bodyData, data };
+
+  if (typeof data === 'object' && !isEmpty(data)) {
+    messageBody = { ...messageBody, data };
+  }
+
+  if (typeof data === 'string' && data !== ' ') {
+    messageBody = { ...messageBody, data: { response: data } };
+  }
+
+  return { ...DEFAULT_MESSAGE_BODY, ...messageBody };
+};
+
+export const formatMessage = (message: Object): string => {
+  const at = message.date.toISOString();
+  const [context, messageData] = message.data;
+  const { message: msg, data = {}, environmentData } = messageData;
+  const { network, os, platformVersion, version } = environmentData;
+
+  const messageBodyParams: ConstructMessageBodyParams = {
+    at,
+    env: `${network}:${os}:${platformVersion}`,
+    ns: [
```

Comment:
the idea of a namespace is to be able to distinguish log messages created by different subsystems within the application (e.g. 'network', 'transactions'). if one enters a function in a subsystem, then that subsystem's name is appended to the namespace (like a stack; beware of recursive functions). Then, all messages created in that subsystem will be tagged with that namespace.
This can be helpful for support and also for automated log analysis.
I don't know if it is feasible to add this here.

Reply:
@CodiePP are you happy with what @MarcusHurney added now?

Reply:
yes. just a few more points:
* to be a valid JSON log, there may not be '[..] [<timestamp>]' in front of every line. if you try with `jq`, does it fail on such a file?
* the filed `sev` would contain the severity level (e.g. 'error' in the example below)

Reply:
@MarcusHurney can you please make the changes @CodiePP  asked for? 🙏

Reply:
@CodiePP @nikolaglumac 
Ok, I've made the JSON objects start on a new line and set the message `level` as the value of the `sev` key. Here are some screenshots:

`development`
<img width="605" alt="screen shot 2019-02-08 at 10 30 14 am" src="https://user-images.githubusercontent.com/13347886/52498132-bb577e80-2b8c-11e9-93d4-c0b7daf59551.png">

`production`
<img width="1428" alt="screen shot 2019-02-08 at 10 29 55 am" src="https://user-images.githubusercontent.com/13347886/52498153-c90d0400-2b8c-11e9-822c-05cc7c6a4655.png">

Reply:
@MarcusHurney @CodiePP I have remove dthe `[context] [time]` bit from production logs.
Here's the final log format (now 100% matching the cardano one):
<img width="1492" alt="screen shot 2019-02-09 at 22 51 36" src="https://user-images.githubusercontent.com/376611/52526736-7717cb80-2cbd-11e9-9da0-b63fd2f6a433.png">